server/configuration.go

```diff
@@ -186,6 +186,17 @@ func (p *Plugin) OnConfigurationChange() error {
 		return errors.Wrap(err, "failed to ensure demo user")
 	}
 
+	botId, ensureBotError := p.Helpers.EnsureBot(&model.Bot{
```

Comment:
Happy to see `error`  in the `Helpers` interface, but I fear that without a holistic change to apply this across the API and Hooks, we're making it very hard on plugin authors who won't know of the need to distinguish `err` from `ensureBotError` in this case.

Reply:
This is a really hard question. I kinda want to say that we can use `error` because it's an entirely new interface, helpers. But you make a good point. Going to make the other changes and think about this.

Reply:
Maybe we can go with the convention of using `err` for `error`'s and `appErr` for `*model.AppError`? This makes it easier to distinguish the two types.